### PR TITLE
Create prefunded account being an on-chain faucet for integration tests

### PR DESCRIPTION
**Pull request by wojtek-coreum**
This change is [Reviewable](https://reviewable.io/reviews/coreumfoundation/crust/63)

### CODE REVIEW

File: infra/testing/key.go [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package testing
+
+import "github.com/CoreumFoundation/coreum/pkg/types"
+
+// HornOfPlentyPrivKey is used by testing framework to fund accounts required by integration tests
+var HornOfPlentyPrivKey = types.Secp256k1PrivateKey{0x86, 0x2f, 0x17, 0xc3, 0x51, 0x83, 0xbe, 0x2a, 0x2b, 0x5d, 0x5b, 0x5b, 0xb5, 0x53, 0x86, 0xd8, 0xad, 0xc8, 0xde, 0x51, 0xa9, 0x73, 0x3f, 0xb7, 0x7d, 0x72, 0xb9, 0x29, 0x91, 0xb7, 0x2c, 0x60}
```
Comment (dzmitryhil) [inline infra/testing/key.go:6]: I would recommend to keep the mnemonic or hex key as a string here, sine it's more readble and in the future in will be easier to set it as a ENV variable

**Review by dzmitryhil — CHANGES_REQUESTED** [review]

**Review by wojtek-coreum — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/coreumfoundation/crust/63)* status: 0 of 2 files reviewed, 1 unresolved discussion (waiting on @begmaroman, @miladz68, @oliverasistemas, @xlab, and @ysv)

___
*[`infra/testing/key.go` line 6 at r1](https://reviewable.io/reviews/coreumfoundation/crust/63#-N9g0PAx-RfEzcCAtcuX-r1-6:-N9g0vO98rImvu1NhVsc:b-gsbiwd) ([raw file](https://github.com/coreumfoundation/crust/blob/84b4b08c9dd5db86726be34d0ab635e654149238/infra/testing/key.go#L6)):*
<details><summary><i>Previously, dzmitryhil (Dzmitry Hil) wrote…</i></summary><blockquote>

I would recommend to keep the mnemonic or hex key as a string here, sine it's more readble and in the future it will be easier to set it as a ENV variable
</blockquote></details>

I don't know... personally I don't care. Random key is random key, no magic involved. @miladz68 @ysv ? Do you think that storing mnemonics instead of bytes adds any value here?

**Review by wojtek-coreum — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/coreumfoundation/crust/63)* status: 0 of 2 files reviewed, 1 unresolved discussion (waiting on @begmaroman, @miladz68, @oliverasistemas, @xlab, and @ysv)

___
*[`infra/testing/key.go` line 6 at r1](https://reviewable.io/reviews/coreumfoundation/crust/63#-N9g0PAx-RfEzcCAtcuX-r1-6:-N9g1UXvEkk544Gx0dOq:b-j2mv9f) ([raw file](https://github.com/coreumfoundation/crust/blob/84b4b08c9dd5db86726be34d0ab635e654149238/infra/testing/key.go#L6)):*
<details><summary><i>Previously, wojtek-coreum (Wojtek) wrote…</i></summary><blockquote>

I don't know... personally I don't care. Random key is random key, no magic involved. @miladz68 @ysv ? Do you think that storing mnemonics instead of bytes adds any value here?
</blockquote></details>

Storing mnemonics will involve cosmos sdk to build keys out of them

**Review by dzmitryhil — COMMENTED** [review]
> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/coreumfoundation/crust/63)* status: all files reviewed, 1 unresolved discussion (waiting on @begmaroman, @miladz68, @oliverasistemas, @xlab, and @ysv)

**Review by ysv — CHANGES_REQUESTED** [review]
> Reviewed 1 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/coreumfoundation/crust/63)* status: all files reviewed, 3 unresolved discussions (waiting on @begmaroman, @miladz68, @oliverasistemas, @wojtek-coreum, and @xlab)

___
*[`infra/apps/apps.go` line 36 at r1](https://reviewable.io/reviews/coreumfoundation/crust/63#-N9kmt7cC2XXA-30thI5:-N9kmt7cC2XXA-30thI6:b-chs1mi) ([raw file](https://github.com/coreumfoundation/crust/blob/84b4b08c9dd5db86726be34d0ab635e654149238/infra/apps/apps.go#L36)):*
> ```Go
> func (f *Factory) CoredNetwork(name string, numOfValidators int, numOfSentryNodes int) infra.Mode {
> 	network := app.NewNetwork(f.networkConfig)
> 	initialBalance := "1000000000000000000000000000000000000000000000000000000000000000000000" + network.TokenSymbol()
> ```

lets not use number bigger than our expected total supply so we have realistic numbers

Total supply should be enough for testing
___
*[`infra/apps/apps.go` line 41 at r1](https://reviewable.io/reviews/coreumfoundation/crust/63#-N9kmi8t8DkFFHf_JH6b:-N9kmi8t8DkFFHf_JH6c:b-iohmos) ([raw file](https://github.com/coreumfoundation/crust/blob/84b4b08c9dd5db86726be34d0ab635e654149238/infra/apps/apps.go#L41)):*
> ```Go
> 	must.OK(network.FundAccount(cored.BobPrivKey.PubKey(), initialBalance))
> 	must.OK(network.FundAccount(cored.CharliePrivKey.PubKey(), initialBalance))
> 	must.OK(network.FundAccount(testing.HornOfPlentyPrivKey.PubKey(), initialBalance))
> ```

so this account will be funded for all znets not only integration test ?

For now it is fine I think. Later might be changed
___
*[`infra/testing/key.go` line 6 at r1](https://reviewable.io/reviews/coreumfoundation/crust/63#-N9g0PAx-RfEzcCAtcuX-r1-6:-N9km3Yn1uLQ5-vElCun:bsngrgs) ([raw file](https://github.com/coreumfoundation/crust/blob/84b4b08c9dd5db86726be34d0ab635e654149238/infra/testing/key.go#L6)):*
Good point Dmitry

> Do you think that storing mnemonics instead of bytes adds any value here?
yes it makes sense for me

> Storing mnemonics will involve cosmos sdk to build keys out of them 
if it is just about importing some cosmos-sdk and building private key from menmonic than it is fine for me

**Review by wojtek-coreum — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/coreumfoundation/crust/63)* status: all files reviewed, 2 unresolved discussions (waiting on @begmaroman, @miladz68, @oliverasistemas, @wojtek-coreum, @xlab, and @ysv)

___
*[`infra/apps/apps.go` line 36 at r1](https://reviewable.io/reviews/coreumfoundation/crust/63#-N9kmt7cC2XXA-30thI5:-N9kuBlP7tN9fnuBmD7j:b5h0hd0) ([raw file](https://github.com/coreumfoundation/crust/blob/1bba5f8a0f42817d01fb8084ae646390c3afdac4/infra/apps/apps.go#L36)):*
<details><summary><i>Previously, ysv (Yaroslav Savchuk) wrote…</i></summary><blockquote>

lets not use number bigger than our expected total supply so we have realistic numbers

Total supply should be enough for testing
</blockquote></details>

Not sure. First of all we don't have total supply because our token is an inflationary one. So in theory total supply may go to infinity.
That's why some tests (https://github.com/CoreumFoundation/coreum/blob/master/integration-tests/bank/gas.go#L28) operate using really high amounts to be 1000% sure that nothing wrong can happen even if total supply goes wild.

**Review by ysv — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/coreumfoundation/crust/63)* status: all files reviewed, 2 unresolved discussions (waiting on @begmaroman, @miladz68, @oliverasistemas, @wojtek-coreum, and @xlab)

___
*[`infra/apps/apps.go` line 36 at r1](https://reviewable.io/reviews/coreumfoundation/crust/63#-N9kmt7cC2XXA-30thI5:-N9l9iGn5C-Sm9u6jYt1:b-knqqy6) ([raw file](https://github.com/coreumfoundation/crust/blob/1bba5f8a0f42817d01fb8084ae646390c3afdac4/infra/apps/apps.go#L36)):*
<details><summary><i>Previously, wojtek-coreum (Wojtek) wrote…</i></summary><blockquote>

Not sure. First of all we don't have total supply because our token is an inflationary one. So in theory total supply may go to infinity.
That's why some tests (https://github.com/CoreumFoundation/coreum/blob/master/integration-tests/bank/gas.go#L28) operate using really high amounts to be 1000% sure that nothing wrong can happen even if total supply goes wild.
</blockquote></details>

I would still prefer to use some reasonable amount. Comparable to what we will have in prod.
To have realistic tests
___
*[`infra/testing/key.go` line 6 at r1](https://reviewable.io/reviews/coreumfoundation/crust/63#-N9g0PAx-RfEzcCAtcuX-r1-6:-N9l9dAa8M8_3ilWffiV:b-yo56oa) ([raw file](https://github.com/coreumfoundation/crust/blob/1bba5f8a0f42817d01fb8084ae646390c3afdac4/infra/testing/key.go#L6)):*
> Do you think that storing mnemonics instead of bytes adds any value here?

yes it makes sense for me

> Storing mnemonics will involve cosmos sdk to build keys out of them

if it is just about importing some cosmos-sdk and building private key from menmonic than it is fine for me

**Review by wojtek-coreum — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/coreumfoundation/crust/63)* status: 0 of 8 files reviewed, 2 unresolved discussions (waiting on @begmaroman, @dzmitryhil, @miladz68, @oliverasistemas, @xlab, and @ysv)

___
*[`infra/apps/apps.go` line 36 at r1](https://reviewable.io/reviews/coreumfoundation/crust/63#-N9kmt7cC2XXA-30thI5:-N9lGAIuEdGTb3P_cdog:b-896fix) ([raw file](https://github.com/coreumfoundation/crust/blob/1bba5f8a0f42817d01fb8084ae646390c3afdac4/infra/apps/apps.go#L36)):*
<details><summary><i>Previously, ysv (Yaroslav Savchuk) wrote…</i></summary><blockquote>

I would still prefer to use some reasonable amount. Comparable to what we will have in prod.
To have realistic tests
</blockquote></details>

Done.
___
*[`infra/testing/key.go` line 6 at r1](https://reviewable.io/reviews/coreumfoundation/crust/63#-N9g0PAx-RfEzcCAtcuX-r1-6:-N9lGKd3Eio_uAG050Eo:b-896fix) ([raw file](https://github.com/coreumfoundation/crust/blob/1bba5f8a0f42817d01fb8084ae646390c3afdac4/infra/testing/key.go#L6)):*
<details><summary><i>Previously, ysv (Yaroslav Savchuk) wrote…</i></summary><blockquote>

> Do you think that storing mnemonics instead of bytes adds any value here?

yes it makes sense for me

> Storing mnemonics will involve cosmos sdk to build keys out of them

if it is just about importing some cosmos-sdk and building private key from menmonic than it is fine for me

</blockquote></details>

Done.

**Review by dzmitryhil — COMMENTED** [review]
> Reviewed 4 of 8 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/coreumfoundation/crust/63)* status: 4 of 8 files reviewed, 1 unresolved discussion (waiting on @begmaroman, @dzmitryhil, @miladz68, @oliverasistemas, @xlab, and @ysv)

**Review by dzmitryhil — APPROVED** [review]
> Reviewed 4 of 8 files at r2.
> *[Reviewable](https://reviewable.io/reviews/coreumfoundation/crust/63)* status: all files reviewed, 1 unresolved discussion (waiting on @begmaroman, @miladz68, @oliverasistemas, @xlab, and @ysv)

**Review by ysv — APPROVED** [review]
> Reviewed 8 of 8 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/coreumfoundation/crust/63)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @begmaroman, @miladz68, @oliverasistemas, and @xlab)

**Pull request merged**